### PR TITLE
Sphinx

### PR DESCRIPTION
**Pull request by Lexseal**
Transitioned documentation generation from pdoc to Sphinx because it is able to process pyi files as well as play nicely with additional detailed tutorials.

Finally, added a config file and set up readthedocs hosting

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -36,6 +36,9 @@ classifiers=[
   "Topic :: Utilities",
 ]
 
+[tool.setuptools.package-data]
+mplib = ["examples/*.md", "examples/*.png"]  # include all md files in mplib/examples
```
Comment (KolinGuo) [inline pyproject.toml:40]: Seemed like there're no *.md and *.png files under mplib/examples. Is this a typo that can be removed?

**Review by KolinGuo — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -36,6 +36,9 @@ classifiers=[
...
+
```
Comment (KolinGuo) [inline pyproject.toml:41]: Seemed like there're no *.md and *.png files under mplib/examples. Is this a typo that can be removed?

Comment (Lexseal) [reply pyproject.toml:41]: oh yeah I forgot to remove that. will do right now

**Review by KolinGuo — APPROVED** [review]

File: .readthedocs.yaml [hunk 3]
```diff
@@ -10,4 +10,4 @@ python:
     - requirements: docs/requirements.txt
 
 sphinx:
-  configuration: docs/source/conf.py
\ No newline at end of file
+  configurationP: docs/source/conf.py
```
[the remark this answers is not in the document]
Comment (Lexseal) [reply .readthedocs.yaml]: Yeah thanks for catching it

File: docs/source/conf.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+exclude_patterns = []
+
+import os
+import sys
+sys.path.insert(0, os.path.abspath(os.path.join('..', '..', 'mplib')))
+
+# -- Options for HTML output -------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
+
+html_theme = 'alabaster'
```
[the remark this answers is not in the document]
Comment (Lexseal) [reply docs/source/conf.py]: Thanks for changing it it does look nicer haha

File: docs/source/conf.py
[the remark this answers is not in the document]
Comment (Lexseal) [reply docs/source/conf.py:1]: Sorry GitHub didn't give me any notification : ((

**Pull request merged**